src/identity/ldap/ldap.strategy.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import {
+  BadRequestException,
+  Inject,
+  Injectable,
+  InternalServerErrorException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { AuthGuard, PassportStrategy } from '@nestjs/passport';
+import { Request } from 'express';
+import LdapAuth from 'ldapauth-fork';
+import { Strategy, VerifiedCallback } from 'passport-custom';
+
+import authConfiguration, { AuthConfig } from '../../config/auth.config';
+import { NotInDBError } from '../../errors/errors';
+import { ConsoleLoggerService } from '../../logger/console-logger.service';
+import { UsersService } from '../../users/users.service';
+import { Identity } from '../identity.entity';
+import { IdentityService } from '../identity.service';
+import { ProviderType } from '../provider-type.enum';
+import { LdapLoginDto } from './ldap-login.dto';
+
+const LDAP_ERROR_MAP: Record<string, string> = {
+  /* eslint-disable @typescript-eslint/naming-convention */
+  '530': 'Not Permitted to login at this time',
+  '531': 'Not permitted to logon at this workstation',
+  '532': 'Password expired',
+  '533': 'Account disabled',
+  '534': 'Account disabled',
+  '701': 'Account expired',
+  '773': 'User must reset password',
+  '775': 'User account locked',
+  default: 'Invalid username/password',
+  /* eslint-enable @typescript-eslint/naming-convention */
+};
+
+interface LdapPathParameters {
+  ldapIdentifier: string;
+}
+
+@Injectable()
+export class LdapAuthGuard extends AuthGuard('ldap') {}
+
+@Injectable()
+export class LdapStrategy extends PassportStrategy(Strategy, 'ldap') {
```

Comment:
Maybe we could split this constructor to reduce indentation and improve readability.

Reply:
I tried to split it up a bit more. Is that better?

Reply:
yeah, maybe split of the create or update identity part.